testing/sync-test/src/auth.rs

```diff
@@ -226,14 +262,23 @@ impl TestClient {
         log::info!("Doing oauth flow!");
         let mut fxa = FirefoxAccount::with_config(acct.cfg.clone());
         let oauth_uri = fxa.begin_oauth_flow(&[SYNC_SCOPE], "integration_test")?;
-        let redirected_to = acct.execute_oauth_flow(&oauth_uri)?;
-        let final_url = Url::parse(&redirected_to)?;
-        let query_params = final_url
-            .query_pairs()
-            .into_owned()
-            .collect::<HashMap<String, String>>();
+
+        // We either authenticate using the normal oauth_flow
+        // Or we use a pairing flow with a logged in account
+        // Both should work fine in executing the oauth flow
+        let (code, state) = if rand::random() {
+            acct.execute_oauth_pair_flow(&oauth_uri)?
```

Comment:
It took me a while to understand what was happening here, because you would not ordinarily call `begin_oauth_flow` if you were doing a pairing flow. To make it clearer, I suggest reworking it like so:

```
let (code, state) = if rand::random() {
  let pairing_uri = fxa.begin_pairing_flow(...)
  acct.execute_oauth_pair_flow(&pairing_uri)
} else {
  let oauth_uri = fxa.begin_oauth_flow(...)
  ... the normal oauth stuff here ...
}
```